Mashael_Almosallam/week_01/day_04/main.js

```diff
@@ -0,0 +1,67 @@
+console.log("JS Connected!");
+
+//Part 1: Array Algorithms:
+
+/* 1. Define a function sumArray and a function multiplyArray that sums and multiplies (respectively) all the numbers in an array of numbers. 
+      For example, sumArray([1,2,3,4]) should return 10, and multiplyArray([1,2,3,4]) should return 24. */
+var arrayNum = [1, 2, 3, 4]; // why 0
+//Sum
+function sumArray(addArray) {
+    var result = 0;
+    for (var i = 0; i < addArray.length; i++) {
+        result += addArray[i];
+    }
+    return result;
+}
+console.log("The sum of the array", sumArray(arrayNum));
+//Multiply
+function multiplyArray(multArray) {
+    var result = multArray[0];
+    for (var i = 1; i < multArray.length; i++) {
+        result *= multArray[i];
+    }
+    return result;
+}
+console.log("The multiplication of the array", multiplyArray(arrayNum));
+
+
+/* 2. Define a function reverseString that reverses a string. For example, reverseString("jag testar") 
+      should return the string "ratset gaj". */
+function reverseString(str) {
+    var result = "";
+    for (var i = str.length - 1; i >= 0; i--) {
+        result += str[i];
+    }
+    return result;
+}
+console.log(reverseString("jag testar"));
+/* 3. Write a function findLongestWord that takes an array of words and returns the length of the longest word in the array. */
+var strArray = ["Mashael", "Nasser", "Almosallam"];
+function findLongestWord(strLength) {
+    var result = 0;
+    for (var i = 0; i < strLength.length; i++) {
+        result += strLength[i].length;
+    }
+    return result;
+}
+console.log(findLongestWord(strArray));
```

Comment:
Good start Meshael, so `findLongestWord(strLength)` is returning the sum of lengths (of all the words in the array). What we actually want is to find the length of the ***longest*** word, which is `Almosallam` in this case. So the return value should've been ***10***. How would we fix that?